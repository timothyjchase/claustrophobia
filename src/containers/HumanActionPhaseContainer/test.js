import React from 'react'
import { HumanActionPhaseContainer, mapStateToProps } from './index'

describe('<HumanActionPhaseContainer />', () => {
  describe('with scenarioKey = TEST_SCENARIO', () => {
    const props = {
      currentState: {
        scenarioKey: 'TEST_SCENARIO',
        oilForYourLamp: false,
        threatDice: 1,
        demonsInPlay: 2,
      },
      addDemon: jest.fn(),
      placeDemonicMechanismTile: jest.fn(),
      placeTile: jest.fn(),
      completeHumanActionPhase: jest.fn(),
    }
    const component = shallow(<HumanActionPhaseContainer {...props} />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})

describe('mapStateToProps', () => {
  describe('with basic state', () => {
    const state = { current: {} }
    it('should return currentState', () => {
      expect(mapStateToProps(state)).toEqual({ currentState: {} })
    })
  })
})
