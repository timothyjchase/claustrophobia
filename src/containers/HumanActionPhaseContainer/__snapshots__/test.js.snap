// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HumanActionPhaseContainer /> with scenarioKey = TEST_SCENARIO should match snapshot 1`] = `
<HumanActionPhase
  addDemon={[MockFunction]}
  completeHumanActionPhase={[MockFunction]}
  demonsInPlay={2}
  oilForYourLamp={false}
  placeDemonicMechanismTile={[MockFunction]}
  placeTile={[MockFunction]}
  scenarioKey="TEST_SCENARIO"
  threatDice={1}
/>
`;
